desc/objectives/utils.py

```diff
@@ -130,32 +149,57 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
             )
         A = A[unfixed_rows][:, unfixed_idx]
         b = b[unfixed_rows]
+
     unfixed_idx = indices_idx
+    fixed_idx = np.delete(np.arange(xp.size), unfixed_idx)
+
+    # compute x_scale if not provided
+    if x_scale == "auto":
+        if things is None:
+            things = objective.things
+        else:
+            things = [things] if not isinstance(things, list) else things
+            things = [get_instance(things, type(t)) for t in objective.things]
```

Comment:
This logic will not work as intended if you are optimizing multiple things that are of the same type (because `get_instance` always grabs the first instance from the list). However: 

1. I don't think we have any use cases for that yet. 
2. This will only affect the scaling, so as long as both things have similar scale parameters it will be fine.

Reply:
couldn't this be an issue if we're optimizing eg modular coils with large currents + trim coils with small currents?

I was thinking about the use case for having things different from objective.things and I thought we haven't officially supported that for a while? Also after #1043 there won't be any reason to re-use objectives since the compile cache will be shared between instances.

Reply:
In the case of modular + trim coils, they should already be combined into a `MixedCoilSet` in order to compute the total magnetic field, so that's not an issue here. An example of what I mean is if you were optimizing multiple `Equilibrium` or multiple `_Coil` for some reason, which I don't think you should ever be doing (yet). 

It might be uncommon that `things` will be different from `objective.things`, but I still think it's a use case we need to support. For example, you could save an `ObjectiveFunction` that you used in a previous optimization (as an HDF5 output) and then load it later to use on a different initial guess.

Reply:
> It might be uncommon that things will be different from objective.things, but I still think it's a use case we need to support. For example, you could save an ObjectiveFunction that you used in a previous optimization (as an HDF5 output) and then load it later to use on a different initial guess.

im pretty sure that won't work, and i don't know if it ever really did? At the very least we don't test it or document it, so I don't have any expectations that it would or should work.

Reply:
@f0uriest What are you requesting me to do here? This code works for all of our expected use cases. Or are you saying we should *always* use `objective.things` and not allow the user to pass in separate `things`?

Reply:
that would be my vote but its up to you

Reply:
It would be cleaner if we always used `objective.things`, I just want to make sure that won't cause any problems. I still think it is possible to re-use objectives if the user wants to. But if we aren't worried about that, then I'll make that change.

Reply:
I think reusing objectives currently falls under "undefined behavior" in that its not promised to work but may in certain circumstances. And the "correct" way would be to re-assign `objective.things`, not to pass in a separate set of things anyways.

Reply:
OK, I've changed it to always use `objective.things`. And since this only affects the scaling, it's not that big of a deal if someone tries to re-use an objective with different things anyways.